ocs_ci/ocs/ocp.py

```diff
@@ -106,6 +106,34 @@ def exec_oc_cmd(self, command, out_yaml_format=True, secrets=None, **kwargs):
             return yaml.safe_load(out)
         return out
 
+    def exec_oc_debug_cmd(self, node, cmd_list):
+        """
+        Function to execute "oc debug" command on OCP node
+
+        Args:
+            node (str): Node name where the command to be executed
+            cmd_list (list): List of commands eg: ['cmd1', 'cmd2']
+
+        Returns:
+            out (str): Returns output of the executed command/commands
+
+        Raises:
+            CommandFailed: When failure in command execution
+        """
+        # Appending one empty value in list for string manipulation
+        cmd_list.append(' ')
+        cmd = " || echo 'CMD FAILED';".join(cmd_list)
+        debug_cmd = f"debug nodes/{node} -- chroot /host /bin/bash -c \"{cmd}\""
+        out = str(self.exec_oc_cmd(
+            command=debug_cmd, out_yaml_format=False
+        ))
+        if out.__contains__('CMD FAILED'):
+            logging.error(f"{debug_cmd} \n {out}")
+            raise CommandFailed
+        else:
+            logging.info(f"{debug_cmd} \n {out}")
```

Comment:
I think that it's printed already from run_cmd which is executed from exec_oc_cmd.

log.debug(f"Command output: {r.stdout.decode()}")
So I guess no need to print it again here.

Reply:
Without those print not getting the command output, Automation run output without logging.info/error
Success cmd execution
```------------------------------------------------------------------------------- live log call -------------------------------------------------------------------------------
12:51:59 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/rperiyas/9thOct/auth/kubeconfig debug nodes/ip-10-0-132-136.us-east-2.compute.internal -- chroot /host /bin/bash -c "export KUBECONFIG=/home/core/auth/kubeconfig || echo 'CMD FAILED';oc login -u kubeadmin -p jme5S-XVt22-7Bq4u-YNto6 || echo 'CMD FAILED';oc get pods -n openshift-image-registry || echo 'CMD FAILED'; "
12:52:04 - MainThread - ocs_ci.utility.utils - WARNING - Command warning: Starting pod/ip-10-0-132-136us-east-2computeinternal-debug ...
                                                         To use host binaries, run `chroot /host`
                                                         
                                                         Removing debug pod ...
PASSED                                                                                                                                                                [100%]
```
Failed cmd execution
```------------------------------------------------------------------------------- live log call -------------------------------------------------------------------------------
12:53:13 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/rperiyas/9thOct/auth/kubeconfig debug nodes/ip-10-0-132-136.us-east-2.compute.internal -- chroot /host /bin/bash -c "oc get nodes || echo 'CMD FAILED';lsblk -ja || echo 'CMD FAILED'; "
12:53:19 - MainThread - ocs_ci.utility.utils - WARNING - Command warning: Starting pod/ip-10-0-132-136us-east-2computeinternal-debug ...
                                                         To use host binaries, run `chroot /host`
                                                         
                                                         Removing debug pod ...
FAILED                                                                                                                                                                [100%]
----------------------------------------------------------------------------- live log teardown -----------------------------------------------------------------------------
12:53:19 - Dummy-7 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/rperiyas/9thOct/auth/kubeconfig get Pod  -A -o yaml
12:53:19 - Dummy-8 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/rperiyas/9thOct/auth/kubeconfig get StorageClass  -A -o yaml
12:53:19 - Dummy-9 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/rperiyas/9thOct/auth/kubeconfig get CephFileSystem  -A -o yaml
12:53:19 - Dummy-10 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/rperiyas/9thOct/auth/kubeconfig get CephBlockPool  -A -o yaml
12:53:19 - Dummy-1 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/rperiyas/9thOct/auth/kubeconfig get PersistentVolume  -A -o yaml
12:53:19 - Dummy-6 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/rperiyas/9thOct/auth/kubeconfig get PersistentVolumeClaim  -A -o yaml
12:53:19 - Dummy-5 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/rperiyas/9thOct/auth/kubeconfig get Namespace  -A -o yaml


================================================================================= FAILURES ==================================================================================
________________________________________________________ TestRegistryDeploymentWithOCS.test_registry_data_pull_push _________________________________________________________

self = <tests.e2e.test_registry.TestRegistryDeploymentWithOCS object at 0x7fc551cd9e80>

    @pytest.mark.polarion_id("OCS-1080")
    @scale
    def test_registry_data_pull_push(self):
        """
        Test case to validate rebooting a node where mgr is running
        should not delete the data collected on prometheus pod
        """
        # helpers.refresh_login_connection()
        # registry.enable_route_and_create_ca_for_registry_access()
        # registry.add_role_to_user(role_type='registry-viewer', user=config.RUN['username'])
        # registry.add_role_to_user(role_type='registry-editor', user=config.RUN['username'])
        # registry.add_role_to_user(role_type='system:registry', user=config.RUN['username'])
        # registry.add_role_to_user(role_type='admin', user=config.RUN['username'])
        # registry.add_role_to_user(role_type='system:image-builder', user=config.RUN['username'])
        # ocp_obj = ocp.OCP()
        # read_only_cmd = f"set env deployment.apps/image-registry REGISTRY_STORAGE_MAINTENANCE_READONLY- " \
        #     f"-n {constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE}"
        # ocp_obj.exec_oc_cmd(read_only_cmd)
        # registry.login_to_podman_kubeadmin_user()
        # logger.info(f"Initial image list")
        # registry.image_list_all()
        # registry.image_pull(image_url='docker.io/library/busybox')
        # logger.info(f"Image list after podman pull")
        # registry.image_list_all()
        # registry.image_push(
        #     image_url='docker.io/library/busybox', namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE
        # )
        # registry.image_list_all()
    
        # cmd = ['export KUBECONFIG=/home/core/auth/kubeconfig', 'oc login -u kubeadmin -p jme5S-XVt22-7Bq4u-YNto6',
        #        'oc get pods -n openshift-image-registry']
        cmd = ['oc get nodes', 'lsblk -ja']
        # helpers.run_debug_cmd(node='ip-10-0-132-136.us-east-2.compute.internal', cmd_list=cmd)
        oc_obj = ocp.OCP()
>       oc_obj.exec_oc_debug_cmd(node='ip-10-0-132-136.us-east-2.compute.internal', cmd_list=cmd)

tests/e2e/test_registry.py:52: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <ocs_ci.ocs.ocp.OCP object at 0x7fc551c70278>, node = 'ip-10-0-132-136.us-east-2.compute.internal', cmd_list = ['oc get nodes', 'lsblk -ja', ' ']

    def exec_oc_debug_cmd(self, node, cmd_list):
        """
        Function to execute "oc debug" command on OCP node
    
        Args:
            node (str): Node name where the command to be executed
            cmd_list (list): List of commands eg: ['cmd1', 'cmd2']
    
        Returns:
            out (str): Returns output of the executed command/commands
    
        Raises:
            CommandFailed: When failure in command execution
        """
        # Appending one empty value in list for string manipulation
        cmd_list.append(' ')
        err_msg = 'CMD FAILED'
        cmd = f" || echo '{err_msg}';".join(cmd_list)
        debug_cmd = f"debug nodes/{node} -- chroot /host /bin/bash -c \"{cmd}\""
        out = str(self.exec_oc_cmd(
            command=debug_cmd, out_yaml_format=False
        ))
        if err_msg in out:
            # logging.error(f"{debug_cmd} \n {out}")
>           raise CommandFailed
E           ocs_ci.ocs.exceptions.CommandFailed

ocs_ci/ocs/ocp.py:133: CommandFailed
========================================================================= 1 failed in 40.97 seconds =========================================================================
```

Reply:
Do you mean that we don't print the debug to the console output when executing run-ci? We don't by default but for local execution I guess you can change by `--log-cli-level` pytest parameter ([doc](https://docs.pytest.org/en/latest/logging.html#live-logs)).  The debug level should be in the log file we produce for each test if I am not wrong. Can you check also log file if it's reported there twice once with debug and second with your info?

Reply:
@petr-balogh Yes, by using --log-cli-level=DEBUG it's getting printed, with existing print in code it is printed twice as you mentioned. 
But in-case of INFO without print, output will not be there in logs. Is this ok ? will remove `logging.error` and `logging.info` from the code.
```
------------------------------------------------------------------------------- live log call -------------------------------------------------------------------------------
14:16:31 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/rperiyas/9thOct/auth/kubeconfig debug nodes/ip-10-0-132-136.us-east-2.compute.internal -- chroot /host /bin/bash -c "oc get nodes || echo 'CMD FAILED';lsblk -ja || echo 'CMD FAILED'; "
14:16:36 - MainThread - ocs_ci.utility.utils - DEBUG - Command output: error: Missing or incomplete configuration info.  Please login or point to an existing, complete config file:
                                                       
                                                         1. Via the command-line flag --config
                                                         2. Via the KUBECONFIG environment variable
                                                         3. In your home directory as ~/.kube/config
                                                       
                                                       To view or setup config directly use the 'config' command.
                                                       CMD FAILED
                                                       lsblk: invalid option -- 'j'
                                                       Try 'lsblk --help' for more information.
                                                       CMD FAILED
14:16:36 - MainThread - ocs_ci.utility.utils - WARNING - Command warning: Starting pod/ip-10-0-132-136us-east-2computeinternal-debug ...
                                                         To use host binaries, run `chroot /host`
                                                         
                                                         Removing debug pod ...
FAILED                                                                                                                                                                [100%]
----------------------------------------------------------------------------- live log teardown -----------------------------------------------------------------------------```
But for --log-cli-level=INFO not getting the output.
```
------------------------------------------------------------------------------- live log call -------------------------------------------------------------------------------
18:07:57 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/rperiyas/9thOct/auth/kubeconfig debug nodes/ip-10-0-132-136.us-east-2.compute.internal -- chroot /host /bin/bash -c " export KUBECONFIG=/home/core/auth/kubeconfig || echo 'cmd failed with exit code \$?';oc login -u kubeadmin -p jme5S-XVt22-7Bq4u-YNto6 || echo 'cmd failed with exit code \$?';oc get pods -n openshift-image-registry "
18:08:03 - MainThread - ocs_ci.utility.utils - WARNING - Command warning: Starting pod/ip-10-0-132-136us-east-2computeinternal-debug ...
                                                         To use host binaries, run `chroot /host`
                                                         
                                                         Removing debug pod ...
FAILED                                                                                                                                                                [100%]
----------------------------------------------------------------------------- live log teardown -----------------------------------------------------------------------------```

Reply:
Removed the print statements in the code.